misc/multistream-select/src/tests.rs

```diff
@@ -244,4 +242,37 @@ fn select_proto_serial() {
 
     async_std::task::block_on(run(Version::V1));
     async_std::task::block_on(run(Version::V1Lazy));
+    async_std::task::block_on(run(Version::V1SimultaneousOpen));
+}
+
+#[test]
+fn simultaneous_open() {
+    async fn run(version: Version) {
+        let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+        let listener_addr = listener.local_addr().unwrap();
+
+        let server = async move {
+            let connec = listener.accept().await.unwrap().0;
+            let protos = vec![b"/proto1", b"/proto2"];
+            let (proto, io, _) = dialer_select_proto_serial(connec, protos, version)
+                .await
+                .unwrap();
+            assert_eq!(proto, b"/proto2");
+            io.complete().await.unwrap();
+        };
+
+        let client = async move {
+            let connec = TcpStream::connect(&listener_addr).await.unwrap();
+            let protos = vec![b"/proto3", b"/proto2"];
+            let (proto, io, _) = dialer_select_proto_serial(connec, protos.into_iter(), version)
+                .await
+                .unwrap();
+            assert_eq!(proto, b"/proto2");
+            io.complete().await.unwrap();
+        };
+
+        futures::future::join(server, client).await;
+    }
```

Comment:
This could just be an `async move` block inside `block_on` unless we want to add more tests?